.github/workflows/cpu_tests.yml

```diff
@@ -85,12 +85,20 @@ jobs:
     runs-on: ubuntu-latest
     needs: [pre_job, docker_update]
     strategy:
-      matrix: ${{fromJson(needs.pre_job.outputs.matrix)}}
+      matrix:
+        toolchain: ${{fromJson(needs.pre_job.outputs.toolchain_matrix)}}
+        compiler: ['gnu', 'clang']
```

Comment:
Small suggestion

```suggestion
        compilers:
          - c: gcc
            cxx: g++
          - c: clang
            cxx: clang++
```

Reply:
I didn't realise you could create sub-sets like this. So the env command below would become:
```
    env:
      CC: ${{ matrix.compiler.c }}
      CXX: ${{ matrix.compiler.cxx }}
```
?

Reply:
![image](https://github.com/user-attachments/assets/4be19437-2328-4927-a1c6-482c924f01b2)
Are you sure this is possible?

Reply:
Maybe I did a mistake, we have something like this in DDC https://github.com/CExA-project/ddc/blob/35de9ca101f1639b0fa9d516ad7aa3363566a9fc/.github/workflows/tests-ubuntu.yaml#L201-L224

Reply:
Strange the editor highlights it as buggy but it doesn't seem to actually cause any problems:
[154e8f5](https://github.com/gyselax/gyselalibxx/pull/275/commits/154e8f5816863a285ba9b66c7364874d15701a4f)

Reply:
Indeed this is the first time I also notice this. I remember Paul Z. telling me the support is not really official.